### PR TITLE
Adding MapDB processor state and using it in Service Map Stateful Processor

### PR DESCRIPTION
**Pull request by AustinTag**
*Issue #, if available:*

*Description of changes:*

- Adding MapDB processor state plugin
- Using this plugin in the service map stateful processor
- Adding new, more generic, interface method for iterating over segments in processor state

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-plugins/lmdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/LmdbProcessorState.java [hunk 1]
```diff
@@ -145,15 +145,30 @@ public long size() {
         }
     }
 
+    private long[] getRange(int segments, int index) {
```
Comment (erosas) [inline situp-plugins/lmdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/LmdbProcessorState.java:148]: you should return some sort of pair, I know Java is not the cleanest for this but the array is not intuitive

**Review by erosas — APPROVED** [review]

File: situp-plugins/mapdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/MapDbProcessorState.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    @Override
+    public void clear() {
+        map.clear();
+    }
+
+    @Override
+    public void close() {
+        map.close();
+    }
```
Comment (kowshikn) [inline situp-plugins/mapdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/MapDbProcessorState.java:117]: nit: From the first version of interface I have some odd feelings on this `clear` and `close`. I think we should just support one interface which is `delete`.  Delete works for any state implementation. Also, clear is expensive and close doesn't define cleaning up of resource. I agree mapdb support deleteOnClose, but other implementations won't.

File: situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java [hunk 3]
```diff
@@ -225,10 +226,10 @@ private String getTraceGroupName(final byte[] traceId) {
      * Delete current state held in the processor
      */
     public void deleteState() {
```
Comment (kowshikn) [inline situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java:228]: This is not used, can we remove this?

File: situp-plugins/common/src/test/java/com/amazon/situp/plugins/processor/state/ProcessorStateTest.java [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
-import java.util.UUID;
-import java.util.function.BiFunction;
-import java.util.stream.Collectors;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 
+import java.io.Serializable;
+import java.util.*;
```
Comment (kowshikn) [inline situp-plugins/common/src/test/java/com/amazon/situp/plugins/processor/state/ProcessorStateTest.java:10]: nit: Avoid *.

File: situp-plugins/lmdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/LmdbProcessorState.java (hunk 1 above)
Comment (kowshikn) [reply situp-plugins/lmdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/LmdbProcessorState.java:148]: ^ +1, maybe something like `KeyRange` object

Comment (AustinTag) [reply situp-plugins/lmdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/LmdbProcessorState.java:148]: Good point, ill make a KeyRange object. Pair requires javafx, which i'd rather not pull in if i dont have to

File: situp-plugins/mapdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/MapDbProcessorState.java (hunk 2 above)
Comment (AustinTag) [reply situp-plugins/mapdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/MapDbProcessorState.java:117]: That makes sense. I'll merge the interfaces into delete()

File: situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java (hunk 3 above)
Comment (AustinTag) [reply situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java:228]: I was using it in the test class, but I can just manually clean the directory

File: situp-plugins/common/src/test/java/com/amazon/situp/plugins/processor/state/ProcessorStateTest.java (hunk 4 above)
Comment (AustinTag) [reply situp-plugins/common/src/test/java/com/amazon/situp/plugins/processor/state/ProcessorStateTest.java:10]: oops, yeah that was auto import. I will remove it

**Review by kowshikn — APPROVED** [review]

**Pull request merged**